modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_file_entry.jsp

```diff
@@ -444,9 +444,13 @@ renderResponse.setTitle(headerTitle);
 										com.liferay.dynamic.data.mapping.storage.DDMFormValues ddmFormValues = null;
 
 										try {
-											DLFileEntryMetadata fileEntryMetadata = DLFileEntryMetadataLocalServiceUtil.getFileEntryMetadata(ddmStructure.getStructureId(), fileVersionId);
+											ddmFormValues = dlEditFileEntryDisplayContext.getDDMFormValues(ddmStructure);
 
-											ddmFormValues = dlEditFileEntryDisplayContext.getDDMFormValues(fileEntryMetadata.getDDMStorageId());
+											if (ddmFormValues == null) {
+												DLFileEntryMetadata fileEntryMetadata = DLFileEntryMetadataLocalServiceUtil.getFileEntryMetadata(ddmStructure.getStructureId(), fileVersionId);
+
+												ddmFormValues = dlEditFileEntryDisplayContext.getDDMFormValues(fileEntryMetadata.getDDMStorageId());
+											}
```

Comment:
Can you further simplify this by moving the null check into the DLEditFileEntryDisplayContext#getDDMFormValues? Imo, it would be better to keep all the logic in the display context.

Reply:
@adolfopa The code was updated based on your feedback by @TaiPhamLR. Thanks!